Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -779,6 +779,83 @@ public struct Timestamp: BSONValue, Equatable, Codable {
     }
 }
 
+enum BSONEqualsError: Error { case InvalidArrayArgument(String) }
+
+// See https://github.com/realm/SwiftLint/issues/461
+// swiftlint:disable cyclomatic_complexity
+/**
+ *  A helper function to test equality between two BSONValues. This function tests for exact BSON equality.
+ *  This means that differing types with equivalent value are not equivalent.
+ *
+ *  e.g.
+ *      4.0 (Double) != 4 (Int)
+ *
+ *  NOTE: This function will always return false if it is used with two arrays that are not of the type [BSONValue] or
+ *  [BSONValue?], because any arrays that are not of these types are not valid `BSONValue`'s.
+ *
+ *  * - Parameters:
+ *   - lhs: The left-hand-side BSONValue to compare.
+ *   - rhs: The right-hand-side BSONValue to compare.
+ *
+ * - Returns: True if lhs is equal to rhs, false otherwise.
+ */
+func bsonEquals(lhs: BSONValue, rhs: BSONValue) -> Bool {
```

Comment:
Do these actually need to use named parameters when calling `bsonEquals()`? I think we can prefix these with `_` and allow users to rely on their position without having to specify `lhs` and `rhs`.

Reply:
Sure, done.